// eslint-config-makeapps/index.js
module.exports = {
  parser: 'babel-eslint',
  plugins: [
    'lodash-fp',
    'better',
    'fp',
  ],
  extends: [
    'airbnb-base',
  ],
  parserOptions: {
    ecmaVersion: 2017,
    sourceType: 'module',
  },
  rules: {
    'arrow-parens': ['error', 'as-needed'],
    'better/no-deletes': 'error',
    'better/no-fors': 'error',
    'better/no-instanceofs': 'error',
    'better/no-switches': 'off',
    'better/no-whiles': 'error',
    curly: ['error', 'multi-line'],
    'fp/no-arguments': 'error',
    'fp/no-mutating-assign': 'error',
    'fp/no-mutation': ['error', {
      commonjs: true,
    }],
    'func-names': ['error', 'as-needed'],
    'func-style': [
      'error',
      'declaration',
      { allowArrowFunctions: true },
    ],
    'import/first': 'off',
    'max-len': 'off',
    'no-confusing-arrow': 'off',
    'no-console': 'off',
    'no-underscore-dangle': 'off',
    'no-use-before-define': ['error', 'nofunc'],
    quotes: ['error', 'backtick'],
    'sort-imports': 'error',
    'space-before-function-paren': 'off',
  },
};
